pkg/usagemetrics/googleanalytics.go

```diff
@@ -0,0 +1,79 @@
+package usagemetrics
+
+import (
+	"github.com/golang/glog"
+	analytics "github.com/jpillora/go-ogle-analytics"
+	k8sapi "github.com/openebs/maya/pkg/client/k8s/v1alpha1"
+	openebsversion "github.com/openebs/maya/pkg/version"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sversion "k8s.io/client-go/pkg/version"
+)
+
+const (
+	// GAclientID is the unique code of the project in Google Analytics
+	GAclientID = "UA-127388617-1"
+)
+
+// Event is a custom metric collected by OpenEBS developers
+type Event struct {
+	category string
+	action   string
+	label    string
+	// LabelSet bool
+	value int64
+	// valueSet bool
+}
+
+// NewMetric returns an Event struct with eventCategory, eventAction,
+// eventLabel, eventValue fields
+func NewMetric(c, a, l string, v int64) *Event {
```

Comment:
The name of the func can be `New`.

Reply:
I am in favour of naming this one `NewEvent` as this is accepting values for a hit type of `event`. In future we might use some other types of events, like `timing` or `transaction` or `exception` this way we can create newer `func` as `NewException` or `NewTiming`

Reply:
I am in favour of naming this one NewEvent as this is accepting values for a hit type of event. In future we might use some other types of events, like timing or transaction or exception this way we can create newer func as NewException or NewTiming

The only problem with this naming is that the [library](https://github.com/jpillora/go-ogle-analytics) uses the same convention. I think it will be better if I just use their methods instead of re-writing them again. I had to do this because the library's fields are not exported so it will not be possible to modify them from my code if and when required.